Buildings/Controls/OBC/CDL/Integers/SequenceBinary.mo

```diff
@@ -0,0 +1,125 @@
+within Buildings.Controls.OBC.CDL.Integers;
+block SequenceBinary "Output total stages that should be ON"
+
+  parameter Integer nSta(
+    final min=1)
+    "Maximum stages that could be ON";
+  parameter Real minStaOn(
+    final quantity="Time",
+    final unit="s")
+    "Minimum stage ON duration";
+  parameter Real minStaOff(
+    final quantity="Time",
+    final unit="s")=minStaOn
+    "Minimum stage OFF duration";
+  parameter Real h
+    "Hysteresis for comparing input with threshold";
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput u(
+    final min=0,
+    final max=1)
+    "Real input for specifying stages"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Interfaces.IntegerOutput y
+    "Total number of stages that should be ON"
+    annotation (Placement(transformation(extent={{100,-20},{140,20}}),
+        iconTransformation(extent={{100,-20},{140,20}})));
+
+protected
+  final parameter Real staInt = 1/nSta
+    "Stage interval";
+  final parameter Real staThr[nSta]= {(i-1)*staInt+h for i in 1:nSta}
+    "Stage thresholds";
+  Buildings.Controls.OBC.CDL.Routing.RealScalarReplicator reaScaRep(
+    final nout=nSta)
+    "Replicate input"
+    annotation (Placement(transformation(extent={{-90,-10},{-70,10}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterThreshold greThr[nSta](
```

Comment:
Do we really need `nSta` instances of this block? Why not using one `Greater` and set the input to avoid zero crossing functions.